src/helpers/job.coffee

```diff
@@ -66,12 +67,16 @@ JobListenerConfig = (checkIntervals, checkRepeats) ->
       # whenever this job status is updated, emit the status for this wrapper
       updateJobStatus = @_updateJobStatusFor(id)
       JobStore.on("job.#{jobId}.*", updateJobStatus)
-      JobStore.off("job.#{jobId}.final", updateJobStatus)
+      JobStore.once("job.#{jobId}.final", @stopListening(updateJobStatus))
 
     stopChecking: (id) ->
       jobId = @_getLatestJob(id)
       JobActions.stopChecking(jobId) if jobId?
 
+    stopListening: (updateJobStatus) ->
+      (jobData) ->
+        JobStore.off("job.#{jobData.id}.*", updateJobStatus)
+
```

Comment:
@nathanstitt good point, I didn't even really like it when I was writing it, and I wanted to do it like 94d996d but it was being weird so I just went with that for now.  Does 94d996d work?